tests/quadrilateral_gimp_element_test.cc

```diff
@@ -0,0 +1,513 @@
+      REQUIRE(gradsf(14, 0) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(15, 0) == Approx(0.0).epsilon(Tolerance));
+
+      REQUIRE(gradsf(0, 1) == Approx(-0.0455625).epsilon(Tolerance));
+      REQUIRE(gradsf(1, 1) == Approx(-0.403875).epsilon(Tolerance));
+      REQUIRE(gradsf(2, 1) == Approx(0.359).epsilon(Tolerance));
+      REQUIRE(gradsf(3, 1) == Approx(0.0405).epsilon(Tolerance));
+
+      REQUIRE(gradsf(4, 1) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(5, 1) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(6, 1) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(7, 1) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(8, 1) == Approx(-0.0005625).epsilon(Tolerance));
+      REQUIRE(gradsf(9, 1) == Approx(0.0005).epsilon(Tolerance));
+      REQUIRE(gradsf(10, 1) == Approx(6.25e-05).epsilon(Tolerance));
+      REQUIRE(gradsf(11, 1) == Approx(0.044875).epsilon(Tolerance));
+      REQUIRE(gradsf(12, 1) == Approx(0.0050625).epsilon(Tolerance));
+      REQUIRE(gradsf(13, 1) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(14, 1) == Approx(0.0).epsilon(Tolerance));
+      REQUIRE(gradsf(15, 1) == Approx(0.0).epsilon(Tolerance));
+    }
+
+    // Check Jacobian
+    SECTION("Four noded quadrilateral Jacobian with deformation gradient") {
+      Eigen::Matrix<double, 16, Dim> coords;
+      // clang-format off
+      // clang-format off
+      coords << -1., -1.,
+                 1., -1.,
+                 1.,  1.,
+                -1.,  1.,
+                -3., -3.,
+                -1., -3.,
+                 1., -3.,
+                 3., -3.,
+                 3., -1.,
+                 3.,  1.,
+                 3.,  3.,
+                 1.,  3.,
+                -1.,  3.,
+                -3.,  3.,
+                -3.,  1.,
+                -3., -1.;
+      // clang-format on
+      // clang-format on
+
+      Eigen::Matrix<double, Dim, 1> psize;
+      psize.setZero();
+      Eigen::Matrix<double, Dim, 1> defgrad;
+      defgrad.setZero();
+
+      Eigen::Matrix<double, Dim, 1> xi;
+      xi << 0, 0;
+
+      Eigen::Matrix<double, Dim, Dim> jacobian;
+      // clang-format off
+      jacobian << 1.0, 0,
+                 0, 1.0;
+      // clang-format on
+
+      // Get Jacobian
+      auto jac = quad->jacobian(xi, coords, psize, defgrad);
+
+      // Check size of jacobian
+      REQUIRE(jac.size() == jacobian.size());
+
+      // Check Jacobian
+      for (unsigned i = 0; i < Dim; ++i)
+        for (unsigned j = 0; j < Dim; ++j)
+          REQUIRE(jac(i, j) == Approx(jacobian(i, j)).epsilon(Tolerance));
+    }
+
+    // Check BMatrix with deformation gradient
+    SECTION(
+        "Four noded quadrilateral B-matrix cell with deformation gradient") {
+      // Reference coordinates
+      Eigen::Matrix<double, Dim, 1> xi;
+      xi << 0.5, 0.5;
+
+      // Nodal coordinates
+      Eigen::Matrix<double, 16, Dim> coords;
+      // clang-format off
+      coords << -1., -1.,
```

Comment:
Maybe you can try on a non-unit length cell so the B-matrix is actually scaled, so try similar to the other element tests, so you get 2 * gradsf as your results